homeassistant/components/aemet/weather_update_coordinator.py

```diff
@@ -0,0 +1,655 @@
+            town_name = self._get_town_name()
+            wind_bearing = self._get_wind_bearing(day, hour)
+            wind_max_speed = self._get_wind_max_speed(day, hour)
+            wind_speed = self._get_wind_speed(day, hour)
+
+        # Overwrite weather values with closest station data (if present)
+        if station_data:
+            if AEMET_ATTR_STATION_DATE in station_data:
+                station_dt = dt_util.parse_datetime(
+                    station_data[AEMET_ATTR_STATION_DATE] + "Z"
+                )
+                station_timestamp = dt_util.as_utc(station_dt)
+            if AEMET_ATTR_STATION_HUMIDITY in station_data:
+                humidity = format_float(station_data[AEMET_ATTR_STATION_HUMIDITY])
+            if AEMET_ATTR_STATION_PRESSURE_SEA in station_data:
+                pressure = format_float(station_data[AEMET_ATTR_STATION_PRESSURE_SEA])
+            if AEMET_ATTR_STATION_TEMPERATURE in station_data:
+                temperature = format_float(station_data[AEMET_ATTR_STATION_TEMPERATURE])
+            if AEMET_ATTR_STATION_TEMPERATURE_RELATIVE in station_data:
+                temperature_feeling = format_float(
+                    station_data[AEMET_ATTR_STATION_TEMPERATURE_RELATIVE]
+                )
+
+        # Get forecast from weather data
+        forecast_weather = self._get_forecast_from_weather_response(weather_response)
+
+        return {
+            ATTR_API_CONDITION: condition,
+            ATTR_API_FORECAST: forecast_weather,
+            ATTR_API_HUMIDITY: humidity,
+            ATTR_API_TEMPERATURE: temperature,
+            ATTR_API_TEMPERATURE_FEELING: temperature_feeling,
+            ATTR_API_PRESSURE: pressure,
+            ATTR_API_RAIN: rain,
+            ATTR_API_RAIN_PROB: rain_prob,
+            ATTR_API_SNOW: snow,
+            ATTR_API_SNOW_PROB: snow_prob,
+            ATTR_API_STATION_ID: station_id,
+            ATTR_API_STATION_NAME: station_name,
+            ATTR_API_STATION_TIMESTAMP: station_timestamp,
+            ATTR_API_STORM_PROB: storm_prob,
+            ATTR_API_TOWN_ID: town_id,
+            ATTR_API_TOWN_NAME: town_name,
+            ATTR_API_TOWN_TIMESTAMP: town_timestamp,
+            ATTR_API_WIND_BEARING: wind_bearing,
+            ATTR_API_WIND_MAX_SPEED: wind_max_speed,
+            ATTR_API_WIND_SPEED: wind_speed,
+        }
+
+    def _get_forecast_from_weather_response(self, weather_response):
+        if self._forecast_mode == FORECAST_MODE_DAILY:
+            forecast = self._get_daily_forecast_from_weather_response(weather_response)
+        elif self._forecast_mode == FORECAST_MODE_HOURLY:
+            forecast = self._get_hourly_forecast_from_weather_response(weather_response)
+        else:
+            forecast = None
+        return forecast
+
+    def _get_daily_forecast_from_weather_response(self, weather_response):
+        if weather_response.daily:
+            parse = False
+            now = dt_util.now()
```

Comment:
This is a minor nit. I would pass in the timestamp to avoid fetching it more than once. Makes it a bit more deterministic.